foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/expressions/ExtractOperator.java

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     07/29/2022-4.0.0 Tomas Kraus - 1573: Fixed types returned by JPQL EXTRACT()
+package org.eclipse.persistence.expressions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.persistence.internal.expressions.ExpressionJavaPrinter;
+import org.eclipse.persistence.internal.expressions.ExpressionSQLPrinter;
+import org.eclipse.persistence.internal.expressions.LiteralExpression;
+import org.eclipse.persistence.internal.helper.ClassConstants;
+
+/**
+ * Expression operator customization for {@code EXTRACT(<date-time-part> FROM <date-time>)}.
+ * Contains support for platform specific native SQL override for individual {@code <date-time-part>}
+ * values.
+ */
+public class ExtractOperator extends ExpressionOperator {
```

Comment:
This should not be public class in an official API - either lower the visibility or move it to the `internal` package

Reply:
Can't lower visibility so moving to internal.